doc/latch.qbk

```diff
@@ -0,0 +1,164 @@
+[/
+  Copyright 2023 Christian Mazakas
+  Distributed under the Boost Software License, Version 1.0.
+  https://boost.org/LICENSE_1_0.txt
+]
+
+[section:latch latch]
+
+[simplesect Authors]
+
+* Christian Mazakas
+
+[endsimplesect]
+
+[section Header <boost/core/latch.hpp>]
+
+The header `<boost/core/latch.hpp>` implements, in a portable way,
+the C++20 `<latch>` header.
+
+`latch` is a single-use barrier which counts downwards, enabling synchronization
+between a set of threads. The counter can be manually decremented by any value,
+but decrementing below zero produces UB. The maximum number of waiters is
+specified to be `boost::core::latch::max()`.
+
+[endsect]
+
+[section Example]
+
+```
+std::ptrdiff_t const num_threads = 16;
+boost::core::latch l(num_threads);
+
+std::vector<std::thread> threads;
```

Comment:
It will be better if the example compiles instead of being a fragment. So it would be better if this all is in a function.